gateway-service/src/main/java/org/zowe/apiml/gateway/conformance/ValidateAPIController.java

```diff
@@ -10,96 +10,164 @@
-import org.springframework.http.HttpStatus;
-import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.*;
+import org.zowe.apiml.message.api.ApiMessage;
+import org.zowe.apiml.message.api.ApiMessageView;
 import org.zowe.apiml.message.core.MessageService;
 
-import io.swagger.v3.oas.annotations.parameters.RequestBody;
-import lombok.RequiredArgsConstructor;
+import java.util.ArrayList;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 /**
- * Controller offered methods for validating serviceID under conformance criteria, it offer methods to
- * check the validation of the given serviceID
+ * Controller offers method to check the validation of the given serviceID under conformance criteria
  */
-@Controller
+@RestController
 @RequiredArgsConstructor
 public class ValidateAPIController {
 
-    private static final Pattern symbolPattern = Pattern.compile("[^a-z0-9]");
-    private static final Predicate<String> isTooLong = serviceId -> (serviceId).length() <= 64;
+    private static final int MaximumServiceIdLength = 64;
+    private static final String InvalidServiceIdRegex = "[^a-z0-9]";
+
+
+    private static final String wrongServiceIdKey = "org.zowe.apiml.gateway.verifier.wrongServiceId";
+    private static final String NoMetadataKey = "org.zowe.apiml.gateway.verifier.NoMetadata";
+    private static final String NonConformantKey = "org.zowe.apiml.gateway.verifier.NonConformant";
+
+
     private final MessageService messageService;
 
     private final VerificationOnboardService verificationOnboardService;
-    private String invalidKey = "org.zowe.apiml.gateway.verifier.wrongServiceId";
+
 
     /**
-     * Accept serviceID and return the JSON file with appropirate message to show if it is valid
-     *
-     * @param serviceID accepted serviceID to check validation
-     * @return return the JSON file message of whether the serviceID is valid
+     * Accepts serviceID and checks conformance criteria
      *
+     * @param serviceId accepted serviceID to check for conformance
+     * @return 200 if service is conformant, 400 + JSON explanation if not
      */
-    @PostMapping(
-        value = "/validate",
+    @GetMapping(
+        value = "/gateway/conformance/{serviceId}",
         produces = MediaType.APPLICATION_JSON_VALUE
     )
-    public ResponseEntity<Object> checkValidate(@RequestBody String serviceID) {
+    @HystrixCommand
+    public ResponseEntity<String> checkConformance(@PathVariable String serviceId) {
+
+
+        ConformanceProblemsContainer foundNonConformanceIssues = new ConformanceProblemsContainer();
 
-        String message = validator(serviceID);
 
-        if (!message.isEmpty()) {
-            return new ResponseEntity<>(messageService.createMessage(invalidKey, message).mapToApiMessage(), HttpStatus.BAD_REQUEST);
+        foundNonConformanceIssues.put("Registration problems", validatorItem1(serviceId));
```

Comment:
Like discussed, we will verify the other naming standard criteria first and then  the registration by calling discoveryClient.